File: daily/654-Maximum-binary-tree.py
class Solution:
    def constructMaximumBinaryTree(self, nums: List[int]) -> Optional[TreeNode]:
        def dfs(arr: List[int]):
            if not arr:
                return None
            max_value = max(arr)
            index_maxVal = arr.index(max_value)
            node = TreeNode(max_value)
            left_prefix = arr[0:index_maxVal]
            right_suffix = arr[index_maxVal + 1: ]
            node.left = dfs(left_prefix)
            node.right = dfs(right_suffix)
            return node

        return dfs(nums)

File: daily/Binary Tree/654-Maximum-binary-tree.go
package daily

func constructMaximumBinaryTree(nums []int) *TreeNode {
	if len(nums) == 0 {
		return nil
	}

	max_index, max_value := 0, 0
	for i := 0; i < len(nums); i++ {
		if nums[i] > max_value {
			max_value = nums[i]
			max_index = i
		}
	}
	node := &TreeNode{}
	node.Val = max_value
	node.Left = constructMaximumBinaryTree(nums[:max_index])
	node.Right = constructMaximumBinaryTree(nums[max_index+1:])
	return node
}
